### ISSUE TITLE
Breaking big transaction into smaller ones.

### ISSUE THREAD

**Issue by levindu** [syncany/syncany#364]
Currently, if there're tons of new files to upload, it will take a lot of time and cache disk space for syncany to do the indexing work before uploading. A unexpected interrupt might lead to the process to restart again.

I'd like to breaking this big transaction into smaller ones. Maybe couple of files or several mega bytes in one transaction. It's easier to  recovery from interruption.

Or streamlining the index and upload of files,  to reduce wait time before real upload.

**Comment by binwiederhier** [comment]
Small correction: Interrupted uploads will not restart, but resume. However, your idea is still valid.

Quick opinion on this: 

- Good: This should be easy enough to implement and it reduces the "commit" size; other clients will be able to retrieve the intermediate state much earlier.
- Bad: Doing that will create a "virtual" state, i.e. a state of the file system that never really existed. That might lead to issues when browsing the history. But I think it's minor.

In short: Good idea. Let's do it.

**Comment by binwiederhier** [comment]
@pimotte Would you like to do this once you finished #345?

**Comment by pimotte** [comment]
Implemented by #437
